### ISSUE TITLE
How does one update the meta-browser to latest version of Chromium (50.X)?

### ISSUE THREAD

**Issue by gbucas** [ds-hwang/wiki#2]
This was VERY useful - thanks ds-hwang - MUCH appreciated.

Your current repository https://github.com/ds-hwang/meta-browser.git
uses Chromium version 40.XXXX.

Could you give us instructions on how we can update that to the latest (or more recent) version of Chromium?

THANKS!

Gerard

**Comment by ds-hwang** [comment]
Hi,

I forked it from https://github.com/OSSystems/meta-browser and didn't do anything.
Meanwhile, the upstream rebased it to M48. https://github.com/OSSystems/meta-browser/commit/37695ddb190d0e39f93d446af11975f8954cb27a

In addition, [meta-crosswalk-embedded](https://github.com/MCherifiOSS/meta-crosswalk-embedded) explains very well how to build and flash yocto image.

**Comment by gbucas** [comment]
Hi DS

Thanks for your fast response - MUCH appreciated..!

I know how to build yocto & flash image. Have done it many times - so no problem there. 

Does your fork 'simply' point to the OSSsystems repository? In other words if I rebuild I will now automatically get M48 (which is good!)?

I tried to build it over the weekend but I keep on getting a build error on libsoup which I can't seem to resolve. Maybe 'master' branch is temporarily broken?

The error I get is:

Any ideas from anyone on how to resolve this?

x86_64-poky-linux-libtool: compile:  x86_64-poky-linux-gcc -m64 -march=core2
-mtune=core2 -msse3 -mfpmath=sse
--sysroot=/home/gerard/development/minnowMax/yocto/poky/build/tmp/sysroots/g
enericx86-64 -DHAVE_CONFIG_H -I. -I../../libsoup-2.54.1/libsoup -I..
-DG_LOG_DOMAIN=\"libsoup\" -DLIBSOUP_COMPILATION
-DLOCALEDIR=\"/usr/share/locale\" -I../../libsoup-2.54.1 -I.. -pthread
-I/home/gerard/development/minnowMax/yocto/poky/build/tmp/sysroots/genericx8
6-64/usr/include/glib-2.0
-I/home/gerard/development/minnowMax/yocto/poky/build/tmp/sysroots/genericx8
6-64/usr/lib/glib-2.0/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_38
-I/home/gerard/development/minnowMax/yocto/poky/build/tmp/sysroots/genericx8
6-64/usr/include/libxml2 -fvisibility=hidden -O2 -pipe -g
-feliminate-unused-debug-types
-fdebug-prefix-map=/home/gerard/development/minnowMax/yocto/poky/build/tmp/w
ork/core2-64-poky-linux/libsoup-2.4/2.54.1-r0=/usr/src/debug/libsoup-2.4/2.5
4.1-r0
-fdebug-prefix-map=/home/gerard/development/minnowMax/yocto/poky/build/tmp/s
ysroots/x86_64-linux=
-fdebug-prefix-map=/home/gerard/development/minnowMax/yocto/poky/build/tmp/s
ysroots/genericx86-64= -Wall -Wstrict-prototypes -Werror=missing-prototypes
-Werror=implicit-function-declaration -Werror=pointer-arith
-Werror=init-self -Werror=format=2 -Wno-format-zero-length
-Werror=missing-include-dirs -Werror=aggregate-return
-Werror=declaration-after-statement -c
../../libsoup-2.54.1/libsoup/soup-session-feature.c  -fPIC -DPIC -o
.libs/libsoup_2_4_la-soup-session-feature.o
make[3]: Leaving directory
`/home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-
linux/libsoup-2.4/2.54.1-r0/build/libsoup'
make[2]: *** [all] Error 2
make[2]: Leaving directory
`/home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-
linux/libsoup-2.4/2.54.1-r0/build/libsoup'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory
`/home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-
linux/libsoup-2.4/2.54.1-r0/build'
make: *** [all] Error 2
WARNING: exit code 1 from a shell command.
ERROR: oe_runmake failed
ERROR: Function failed: do_compile (log file is located at
/home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-l
inux/libsoup-2.4/2.54.1-r0/temp/log.do_compile.31332)

Any ideas on how to fix that!?

Thanks!

Gerard

**Comment by ds-hwang** [comment]
> Does your fork 'simply' point to the OSSsystems repository? In other words if I rebuild I will now automatically get M48 (which is good!)?

yes, because recipes downloads chromium source from [cloud](http://gsdview.appspot.com/chromium-browser-official/${P}.tar.xz)

> I tried to build it over the weekend but I keep on getting a build error on libsoup which I can't seem to resolve. Maybe 'master' branch is temporarily broken?

yocto 'master' branch can be broken anytime. I guess stable branch might work.

> ERROR: Function failed: do_compile (log file is located at
/home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-l
inux/libsoup-2.4/2.54.1-r0/temp/log.do_compile.31332)
> Any ideas on how to fix that!?

I want to see /home/gerard/development/minnowMax/yocto/poky/build/tmp/work/core2-64-poky-l
inux/libsoup-2.4/2.54.1-r0/temp/log.do_compile.31332

- DS

**Comment by gbucas** [comment]
Hi again DS

 

REALLY appreciate your willingness to look at this – THANKS!!!!

 

That file is attached…

 

I also attached my local.conf and bblayers.conf files (just in case..).

 

Regards

 

Gerard

**Comment by ds-hwang** [comment]
Hi @gbucas 

I got your mail somehow but I could not find any attached file?

Br, DS

**Comment by gbucas** [comment]
It was attached – can you send me your “normal” email address?

 

Will resend it

**Comment by gbucas** [comment]
OK - seems github doesn't accept the file extensions I sent you. Here is a zip version!
[libsoupErrorLog.zip](https://github.com/ds-hwang/wiki/files/277806/libsoupErrorLog.zip)

**Comment by gbucas** [comment]
[libsoupErrorLog.zip](https://github.com/ds-hwang/wiki/files/277813/libsoupErrorLog.zip)

**Comment by gbucas** [comment]
Ok – apparently github doesn’t accept the file extensions of log or conf files!

 

So here is a zip version…

 

Gerard

**Comment by ds-hwang** [comment]
for some reason, `gssapi.h` doesn't exist. 
`fatal error: gssapi/gssapi.h: No such file or directory`

need to add the dependency to recipes. I don't know what is the right dependency. maybe libkrb5 or something.

**Comment by gbucas** [comment]
I saw that but couldn't find a good yocto repository for libkrb5 ...

Do you know where to find a yocto 'ready' repository for that..?

I am not THAT good! ;)

**Comment by ds-hwang** [comment]
I don't know. When I need new package, I usually `git grep` in the yocto tree. Maybe there is better way.

how about reporting this issue in  https://github.com/OSSystems/meta-browser ?
They probabaly will fix it.

**Comment by gbucas** [comment]
Ok - let me try that !

**Comment by gbucas** [comment]
FYI ..!

**Comment by gbucas** [comment]
Hi again DS

 

Still need your help!

 

I started from scratch (re-downloaded all master branches) again last night. Have two different issues now:

 

1.     Used your instructions and populated meta-browser from git clone https://github.com/ds-hwang/meta-browser.git.
In that case it still builds the Chromium-40 browser (not V48.X) & I get the attached error when building Chromium.


2.     I then started over again and this time I populated meta-browser directly from git clone 
Then everything built successfully & I got a bootable yocto image with Chromium-48 browser. So that looked good & it booted OK (got Desktop, and Chromium icon and Chromium which starts up OK, etc). The PROBLEM here is that Chromium can’t render any pages. Even something like “Chrome://flags” or “Chrome://settings” won’t render and it simply displays “Aw Shucks – something went wrong with this page” (ALL pages!). So I guess, there is some problem with the configurations of Chromium to rendered (or whatever?)? 

Any ideas what I can do to fix it?



3.     Can you tell me what exactly is the difference between your https://github.com/ds-hwang/meta-browser.git fork and the https://github.com/OSSystems/meta-browser repository (besides the change from M40 to M48)? If there are differences how can I force your fork to upgrade to the M48 branch? It doesn’t seem to do this automatically (it stays at M40 – with some build errors)



Would appreciate some help/advice!

 

Thanks

 

Gerard
